crates/core_arch/src/x86/avx512bf16.rs

```diff
@@ -150,12 +150,11 @@ pub unsafe fn _mm512_maskz_cvtne2ps_pbh (k: __mmask32, a: __m512, b: __m512) ->
 #[cfg_attr(test, assert_instr("vcvtneps2bf16"))]
 pub unsafe fn _mm_cvtneps_pbh (a: __m128) -> __m128bh {
     let mut result:__m128bh;
-    asm!(
-            "vcvtneps2bf16 {0}, {1}",
-            in(xmm_reg) a,
-            lateout(xmm_reg) result,
-            options(att_syntax)
-        );
+    llvm_asm!("vcvtneps2bf16 %xmm0, %xmm0"
+    : "={xmm0}"(result) 
+    : "{xmm0}"(a)
+    : "xmm0" 
+    :);
```

Comment:
Please don't use `llvm_asm!`, it is going to be deprecated soon.

Instead just wrap the `asm!` in a `cfg_if` and provide a dummy `unreachable!()` implementation on other platforms.

Reply:
Thanks for your review. I have updated the code following your suggestions.